### DISCUSSION TITLE
Sooo.... How about an Android Client?

### DISCUSSION THREAD

**Discussion by steelswords** [GothenburgBitFactory/taskwarrior#3686] [Ideas]
Hi there! I *love* Taskwarrior on desktop. It's honestly everything I never knew I needed. But the problem is I'd really like to access it from my Android phone. I know there were one or two very old apps that worked with Taskwarrior 2, but I don't think they've been updated for Taskwarrior 3, so this seems like a great opportunity to build something new for the new and improved Taskwarrior.

Most of the previous discussion on mobile support centers on supporting Caldav and co. I get why that isn't a good fit for Taskwarrior. Mapping back and forth between two largely incompatible formats would be a huge pain. What I'm suggesting instead is building something that works natively with Taskwarrior on mobile. I would get use out of that even if it's not at all fancy. A homescreen widget and an edit activity would be incredibly useful if it syncs to my desktop.

I'm an embedded guy, not an Android dev, but I'm happy to contribute and learn where I can. Any thoughts or interest?

**Comment by linuxcaffe** [comment]
Hi! Try taskwarrior-flutter!

**Comment by felixschurk** [comment]
Hei,
Is taskwarrior-flutter capable of working with the SQLite database?

The whole setup on GitHub looks like taskwarrior 2 setup, with client keys etc.


On 18 Nov 2024 06:37, David J Patrick ***@***.***> wrote:

Hi! Try taskwarrior-flutter!

**Reply by smemsh** [reply]
> Is taskwarrior-flutter capable of working with the SQLite database?

Don't most taskwarrior clients use taskwarrior itself (ie, via the JSON interfaces), not interact directly with the database?

**Comment by linuxcaffe** [comment]
taskwarrior-flutter, as I understand it (dimly) uses an SQL database on android (not sure about other platforms) and is setup to sync via task-server (2.x). It doesn't use task per se, but a local reimplementation of the functions/ algorithms. 

It's still in an early state, limited as compared to full taskwarrior functionality, but it's usable and in active development.

**Reply by steelswords** [reply]
Aha. Looking at [this Github Issue](https://github.com/CCExtractor/taskwarrior-flutter/issues/347) and the [linked documentation](https://its-me-abhishek.github.io/ccsync-docs/) from said issue, it looks like taskwarrior-flutter does now support Taskwarrior v3. Woot! I'll look into that further. Thanks for the tip!

**Comment by linuxcaffe** [comment]
Using task natively on mobile? Ultimate solution there would be a linux phone, using ubuntu-touch, on maybe a Google Pixel 3a, for example. Just like that, your whole CLI world in your hand ;-)

**Reply by steelswords** [reply]
True, there are ways to incorporate access Taskwarrior on my phone with Termux or SSH or something of that nature, but it takes away the whole convenience of using the native Android experience. TUIs, while ideal for me when I have a full keyboard, are just not a great UI for phones. So I think an Android client is still worth pursuing. :)
